mongo/search_index_view.go

```diff
@@ -0,0 +1,258 @@
+// Copyright (C) MongoDB, Inc. 2023-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package mongo
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/writeconcern"
+	"go.mongodb.org/mongo-driver/x/bsonx/bsoncore"
+	"go.mongodb.org/mongo-driver/x/mongo/driver"
+	"go.mongodb.org/mongo-driver/x/mongo/driver/operation"
+	"go.mongodb.org/mongo-driver/x/mongo/driver/session"
+)
+
+// SearchIndexView is a type that can be used to create, drop, list and update search indexes on a collection. A SearchIndexView for
+// a collection can be created by a call to Collection.SearchIndexes().
+type SearchIndexView struct {
+	coll *Collection
+}
+
+// SearchIndexModel represents a new search index to be created.
+type SearchIndexModel struct {
+	// A document describing the definition for the search index. It cannot be nil.
+	// See https://www.mongodb.com/docs/atlas/atlas-search/create-index/ for reference.
+	Definition interface{}
+
+	// The search index options.
+	Options *options.SearchIndexesOptions
+}
+
+// List executes a listSearchIndexes command and returns a cursor over the search indexes in the collection.
+//
+// The name parameter specifies the index name. A nil pointer matches all indexes.
+//
+// The opts parameter can be used to specify options for this operation (see the options.ListSearchIndexesOptions
+// documentation).
+func (siv SearchIndexView) List(ctx context.Context, searchIdxOpts *options.SearchIndexesOptions, opts ...*options.ListSearchIndexesOptions) (*Cursor, error) {
```

Comment:
Optional: Long lines can be difficult to read. Consider wrapping them.

E.g.
```go
func (siv SearchIndexView) List(
	ctx context.Context,
	searchIdxOpts *options.SearchIndexesOptions,
	opts ...*options.ListSearchIndexesOptions,
) (*Cursor, error) {
```
That suggestion applies to all long function signatures in this PR.